File: core/weapons/minature.js
import { worldSize } from "../utils/constants.js";
import { wrap } from "../world/utils.js";
import Projectile from "./projectile.js";

export default class Minature extends Projectile {
    constructor({ x, y, angle, ship, range = 500, color, speed = 10 }) {
        super({
            name: "Minature",
            speed: speed,
            acceleration: speed,
            x: x,
            y: y,
            width: 10,
            height: 10,
            angle: angle,
            damage: 5,
            range,
            fireRate: 1,
            energyCost: 0,
            ship,
            color
        });
    }
}

File: void_front/game/core/weapons/minature.py
from __future__ import annotations

from game.core.weapons.projectile import Projectile


class Minature(Projectile):
    def __init__(self, options, force=False):
        defaults = {
            "name": "Minature",
            "speed": options.get("speed", 10),
            "acceleration": options.get("speed", 10),
            "x": options.get("x"),
            "y": options.get("y"),
            "width": 10,
            "height": 10,
            "angle": options.get("angle"),
            "damage": 5,
            "range": options.get("range", 500),
            "fireRate": 1,
            "energyCost": 0,
            "ship": options.get("ship"),
            "color": options.get("color"),
            "img": options.get("img"),
        }
        super().__init__(defaults, force)
